packages/ui/src/validators/providers/utils.ts

```diff
@@ -123,14 +124,15 @@ export const getValidatorInfo = (
 
           const commission = validator.commission
           const averageReward = rewards.reduce((sum, reward) => sum.add(reward.eraReward), BN_ZERO).divn(rewards.length)
-          const apr =
-            Number(
-              averageReward
-                .muln(ERAS_PER_YEAR)
-                .muln(100 - commission)
-                .muln(100)
-                .div(staking.total)
-            ) / 100
+          const rewardPerEra = averageReward.muln(100 - commission).div(staking.total)
+          const aprValue = rewardPerEra.muln(ERAS_PER_YEAR)
+          const maxSafeValue = new BN(1000000)
+
+          if (aprValue.gt(maxSafeValue)) {
+            return { APR: 1000000 }
+          }
+
+          const apr = aprValue.toNumber()
```

Comment:
It shows 0 for all. The expected value is in %. See the tooltip:
> This is calculated as follow:
> Yearly Reward * (1 - Commission) / Stake
```suggestion
          const rewardPerYear = averageReward.muln(ERAS_PER_YEAR)
          const apr = rewardPerYear.muln(100 - commission).div(staking.total)
```
Does this make sense?